rules/manager.go

```diff
@@ -907,6 +907,34 @@ func (m *Manager) AlertingRules() []*AlertingRule {
 	return alerts
 }
 
+// AlertingRules returns the list of the manager's alerting rules organised by group.
+func (m *Manager) AlertingRulesbyGroup() *AlertingGroups {
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+
+	alerts := &AlertingGroups{}
+	for _, g := range m.groups {
+		group := &AlertingGroup{Name: g.name}
+		for _, rule := range g.rules {
+			if alertingRule, ok := rule.(*AlertingRule); ok {
+				group.Rules = append(group.Rules, alertingRule)
+			}
+		}
+		alerts.Groups = append(alerts.Groups, group)
+	}
+	return alerts
+}
+
+type AlertingGroups struct {
+	Groups []*AlertingGroup
+}
+
+type AlertingGroup struct {
+	Name                 string
+	Rules                []*AlertingRule
+	AlertStateToRowClass map[AlertState]string
```

Comment:
This map is only relevant for display purposes and shouldn't live in the rule code or be attached to every individual group. Best to keep it part of the top-level data value that is passed into the HTML template.